chain/network/src/network_protocol/proto_conv/net.rs

```diff
@@ -3,12 +3,102 @@ use super::*;
 
 use crate::network_protocol::proto;
 use crate::network_protocol::PeerAddr;
-use crate::network_protocol::{Edge, PartialEdgeInfo, PeerInfo};
+use crate::network_protocol::{Edge, PartialEdgeInfo, PeerInfo, OwnedIpAddress, SignedOwnedIpAddress};
 use borsh::{BorshDeserialize as _, BorshSerialize as _};
 use near_primitives::network::AnnounceAccount;
 use protobuf::MessageField as MF;
 use std::net::{IpAddr, SocketAddr};
 
+////////////////////////////////////////
+// Parse std::net::IpAddr to Protocol Buffer and back
+#[derive(thiserror::Error, Debug)]
+pub enum ParseIpAddrError {
+    #[error("invalid IP")]
+    InvalidIP,
+}
+
+impl From<&std::net::IpAddr> for proto::IpAddr{
```

Comment:
The other type is SocketAddr, however, we want to sign on IpAddr instead of SocketAddr, so no point sending the SocketAddr information if all we need is just IpAddr. 

This method is necessary because think of it as like, quicksort implements a partition() function within it. But in order to use the partition() function, you first need to extract that code out into a stand-alone function before it can be re-used for other purposes. 

Socket addr contains IpAddr, but I can't use it's protobuf conversion without first extracting that code out of the SocketAddr implementation.